### PR TITLE
docs: update all pages that use code like text().withText() to the op…

### PR DESCRIPTION
**Pull request by JohannesDienst-askui**
tional parameters docs

### CODE REVIEW

File: docs/docs/api/03-Element-Descriptions/text.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ await aui.click().text('Username').exec();
 await aui.click().text().withExactText('Username').exec();
 
 // Matching with a regex
-await aui.click().text().withTextRegex('\b[Ss]\w+').exec();
+await aui.click().text().withExactText('\b[Ss]\w+').exec();
```
Comment (adi-wan-askui) [inline docs/docs/api/03-Element-Descriptions/text.md:23]: Seems unintended. If it is intended, an you explain?

File: docs/docs/api/03-Element-Descriptions/withtextregex.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ Filters for texts, which match the regex pattern.
 'The rain in switzerland' === withTextRegex('\b[Ss]\w+') => true
 
 // this filters any text that contains 'pie' or 'cake' or 'Pie' or 'Cake'
-await aui.get().text().withTextRegex('.*([Pp]ie|[Cc]ake).*').exec();
+await aui.get().text('.*([Pp]ie|[Cc]ake).*').exec();
```
Comment (adi-wan-askui) [inline docs/docs/api/03-Element-Descriptions/withtextregex.md:17]: Like above, seems unintended

File: docs/docs/general/06-Tutorials/google-cat-search.md [hunk 3]
```diff
@@ -118,7 +118,7 @@ describe('jest with askui', () => {
       .typeIn("cat")
       .textfield()
       .contains()
-      .icon().withText('microphone')
+      .icon('microphone')
```
Comment (adi-wan-askui) [inline docs/docs/general/06-Tutorials/google-cat-search.md:121]: Does that already work? Was not aware of it or forgot about it.

**Review by adi-wan-askui — CHANGES_REQUESTED** [review]
Please split PRs about different things if possible :) Otherwise nice

File: docs/docs/api/03-Element-Descriptions/withtextregex.md
Comment (HaramChoi-askui) [inline docs/docs/api/03-Element-Descriptions/withtextregex.md:1]: Parentheses are missing here`text()` 😄

Comment (JohannesDienst-askui) [reply docs/docs/api/03-Element-Descriptions/withtextregex.md:1]: Refactoring this is not my strong suite it seems...

**Comment by adi-wan-askui** [comment]
It is very difficult to review if you do force-pushes as I cannot filter out the new changes easily. If it is your intention to keep the history clean, please rebase & squash when merging instead.

File: docs/docs/general/06-Tutorials/google-cat-search.md (hunk 3 above)
Comment (adi-wan-askui) [reply docs/docs/general/06-Tutorials/google-cat-search.md:121]: So I interpret the resolving of the thread as "it works"

**Review by adi-wan-askui — APPROVED** [review]

File: docs/docs/general/06-Tutorials/google-cat-search.md (hunk 3 above)
Comment (JohannesDienst-askui) [reply docs/docs/general/06-Tutorials/google-cat-search.md:121]: A mistake from my side on the resolve. Once the issue with relational selectors is fixed I will merge.

**Comment by adi-wan-askui** [comment]
> @JohannesDienst-askui I rebased but did not squash. Seems like I have to brush up on my knowledge 🤔

Only when merging, not before with the open PR, there should be no force push into an open PR necessary

File: docs/docs/general/06-Tutorials/google-cat-search.md (hunk 3 above)
Comment (JohannesDienst-askui) [reply docs/docs/general/06-Tutorials/google-cat-search.md]: With all the different PRs this got mixed up with the `matching()` one. Really sorry for the confusion!

**Comment by adi-wan-askui** [comment]
@JohannesDienst-askui When can this be merged?

**Comment by JohannesDienst-askui** [comment]
I thought I did merge it already. Will merge now!

**Pull request merged**